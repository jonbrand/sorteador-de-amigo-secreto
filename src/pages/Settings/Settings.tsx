import { Footer } from '../../components/Footer/Footer';
import { Form } from '../../components/Form/Form';
import { ListParticipants } from '../../components/ListParticipants/ListParticipants';
import { Card } from '../../components/Card/Card';

export const Settings = () => {
  return (
    <Card>
      <section>
        <h2>
          Vamos Comecar
        </h2>
        <Form />
        <ListParticipants />
        <Footer />
      </section>
    </Card>
  )
}