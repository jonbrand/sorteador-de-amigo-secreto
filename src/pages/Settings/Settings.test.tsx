import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil"
import { Settings } from "./Settings";

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavigate
  }
});

describe('a pagina de configuracao', () => {
  test('deve ser renderizada corretamente', () => {
    const { container } = render(
      <RecoilRoot>
        <Settings />
      </RecoilRoot>
    )

    expect(container).toMatchSnapshot();
  })
})