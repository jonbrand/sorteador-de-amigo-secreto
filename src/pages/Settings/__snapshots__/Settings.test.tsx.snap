// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina de configuracao deve ser renderizada corretamente 1`] = `
<div>
  <div
    class="sc-ftvSup czXKkF"
  >
    <section>
      <h2>
        Vamos Comecar
      </h2>
      <form
        class="sc-hKMtZM"
      >
        <div
          class="sc-eCYdqJ eizkJZ"
        >
          <input
            class="sc-jSMfEi dJsTpL"
            placeholder="Insira os nomes dos participantes"
            type="text"
            value=""
          />
          <button
            class="sc-gKXOVf hIwtvo"
            disabled=""
          >
            Adicionar
          </button>
          
        </div>
      </form>
      <ul />
      <footer
        class="sc-bczRLJ gyGMrX"
      >
        <button
          class="sc-gsnTZi drgOyq"
          disabled=""
        >
          Iniciar brincadeira!
        </button>
        <img
          class="sc-dkzDqf jqkJwu"
          src="bags.svg"
        />
      </footer>
    </section>
  </div>
</div>
`;
